frontend/server/handlers/artifacts.ts

```diff
@@ -239,6 +260,62 @@ function getGCSArtifactHandler(options: { key: string; bucket: string }, peek: n
   };
 }
 
+function getVolumeArtifactsHandler(options: { bucket: string; key: string }, peek: number = 0) {
+  const { key, bucket } = options;
+  return async (req: Request, res: Response) => {
+    const filePath = path.join('/', key);
+    try {
+      // get ml-pipeline-ui pod name
+      const { HOSTNAME: POD_NAME } = process.env;
+      // only check when POD_NAME and serverNamespace can be obtained
+      if (POD_NAME && serverNamespace) {
```

Comment:
nit: can you log a warning message because if either cannot be got, it's an unexpected problem and should better be fixed